v2/tests/log_context_ut/log_context_ut.c

```diff
@@ -0,0 +1,1861 @@
+    
+    LOG_CONTEXT_CREATE_with_one_int32_t_succeeds();
+    LOG_CONTEXT_CREATE_with_two_properties_succeeds();
+    LOG_CONTEXT_CREATE_with_all_int_property_types_succeeds();
+    
+    LOG_CONTEXT_CREATE_with_one_string_property_with_only_format_passed_to_it_succeeds();
+    LOG_CONTEXT_CREATE_with_2_string_properties_with_only_format_passed_to_it_succeeds();
+    LOG_CONTEXT_CREATE_with_a_string_property_using_printf_formatting_succeeds();
+    LOG_CONTEXT_CREATE_with_a_string_property_followed_by_another_int_property_succeeds();
+    LOG_CONTEXT_CREATE_with_a_string_property_preceded_by_another_int_property_succeeds();
+
+    LOG_CONTEXT_CREATE_with_LOG_CONTEXT_NAME_uses_the_context_name();
+
+    LOG_CONTEXT_CREATE_with_a_parent_that_has_no_properties_succeeds();
+    LOG_CONTEXT_CREATE_with_a_parent_that_has_one_int_property_succeeds();
+    LOG_CONTEXT_CREATE_with_a_parent_that_has_2_int_properties_succeeds();
+    LOG_CONTEXT_CREATE_with_a_parent_that_has_a_string_property_succeeds();
+    LOG_CONTEXT_CREATE_with_a_parent_that_has_2_string_properties_succeeds();
+    LOG_CONTEXT_CREATE_with_a_parent_that_has_a_context_name_succeeds();
+
+    LOG_CONTEXT_LOCAL_DEFINE_with_no_properties_succeeds();
+    LOG_CONTEXT_LOCAL_DEFINE_with_1_int_property_succeeds();
+    LOG_CONTEXT_LOCAL_DEFINE_with_2_int_properties_succeeds();
+    LOG_CONTEXT_LOCAL_DEFINE_with_all_int_property_types_succeeds();
+
+    LOG_CONTEXT_LOCAL_DEFINE_with_LOG_CONTEXT_NAME_succeeds();
+
+    LOG_CONTEXT_LOCAL_DEFINE_with_a_parent_that_has_no_properties_succeeds();
+    LOG_CONTEXT_LOCAL_DEFINE_with_a_parent_that_has_one_int_property_succeeds();
+    LOG_CONTEXT_LOCAL_DEFINE_with_a_parent_that_has_2_int_properties_succeeds();
+    LOG_CONTEXT_LOCAL_DEFINE_with_a_parent_that_has_a_string_property_succeeds();
+    LOG_CONTEXT_LOCAL_DEFINE_with_a_parent_that_has_2_string_properties_succeeds();
+    LOG_CONTEXT_LOCAL_DEFINE_with_a_parent_that_has_a_context_name_succeeds();
+
+    LOG_CONTEXT_LOCAL_DEFINE_with_a_stack_log_context_succeeds();
+    LOG_CONTEXT_CREATE_with_a_stack_log_context_succeeds();
+
+    log_context_get_property_value_pair_count_with_NULL_log_context_returns_UINT32_MAX();
+    log_context_get_property_value_pair_count_with_a_local_stack_context_returns_the_count();
+    log_context_get_property_value_pair_count_with_a_dynamically_allocated_context_returns_the_count();
+
+    log_context_get_property_value_pairs_with_NULL_log_context_returns_NULL();
+    log_context_get_property_value_pairs_with_a_local_stack_context_returns_the_pairs();
+    log_context_get_property_value_pairs_with_a_dynamically_allocated_context_returns_the_pairs();
+
+    creating_a_context_with_LOG_MAX_STACK_PROPERTY_VALUE_PAIR_COUNT_minus_one_properties_succeeds();
+    creating_a_context_with_LOG_MAX_STACK_PROPERTY_VALUE_PAIR_COUNT_properties_reports_error();
+    creating_a_context_with_too_much_data_reports_error();
+
+    return asserts_failed;
+}
```

Comment:
I did not have time to review this file! But will sign off to not stop progress 

"Labor Day" is coming!